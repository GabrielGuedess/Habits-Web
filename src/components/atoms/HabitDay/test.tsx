import { render } from '@testing-library/react';

import { HabitDay } from '.';

describe('<HabitDay />', () => {
  it('should render the HabitDay', () => {
    const { container } = render(<HabitDay amount={5} completed={4} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the HabitDay disabled', () => {
    const { container } = render(
      <HabitDay amount={5} completed={4} state="disabled" />,
    );

    expect(container.firstChild).toHaveClass(
      'cursor-not-allowed',
      'opacity-40',
    );
  });

  it('should render the HabitDay with 0 percentage', () => {
    const { container } = render(
      <HabitDay amount={2} completed={0} state="active" />,
    );

    expect(container.firstChild).toHaveClass('border-zinc-800', 'bg-zinc-900');
  });
  it('should render the HabitDay with less 20 percentage', () => {
    const { container } = render(<HabitDay amount={6} completed={1} />);

    expect(container.firstChild).toHaveClass(
      'border-violet-700',
      'bg-violet-900',
    );
  });

  it('should render the HabitDay with less 40 percentage', () => {
    const { container } = render(<HabitDay amount={5} completed={1} />);

    expect(container.firstChild).toHaveClass(
      'border-violet-600',
      'bg-violet-800',
    );
  });

  it('should render the HabitDay with less 60 percentage', () => {
    const { container } = render(<HabitDay amount={11} completed={6} />);

    expect(container.firstChild).toHaveClass(
      'border-violet-500',
      'bg-violet-700',
    );
  });

  it('should render the HabitDay with less 80 percentage', () => {
    const { container } = render(<HabitDay amount={11} completed={8} />);

    expect(container.firstChild).toHaveClass(
      'border-violet-500',
      'bg-violet-600',
    );
  });
});
