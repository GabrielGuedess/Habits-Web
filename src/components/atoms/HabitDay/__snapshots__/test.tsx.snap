// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HabitDay /> should render the HabitDay 1`] = `
<button
  aria-controls="radix-:r0:"
  aria-expanded="false"
  aria-haspopup="dialog"
  class="h-10 w-10 rounded-lg border-2  border-violet-400 bg-violet-500"
  data-state="closed"
  type="button"
/>
`;
